relay-server/src/services/health_check.rs

```diff
@@ -182,127 +208,61 @@ impl HealthCheckService {
+        let check_interval = self.config.health_refresh_interval();
+        // Add 10% buffer to the internal timeouts to avoid race conditions.
+        let status_timeout = (check_interval + self.config.health_probe_timeout()).mul_f64(1.1);
 
         tokio::spawn(async move {
-            let mut shutdown = Controller::shutdown_handle();
-
-            loop {
-                tokio::select! {
-                    biased;
-
-                    Some(message) = rx.recv() => {
-                        let service = service.clone();
-                        tokio::spawn(async move { service.handle_message(message).await });
-                    }
-                    _ = shutdown.notified() => {
-                        service.is_shutting_down.store(true, Ordering::Relaxed);
-                    }
-                }
-            }
-        });
-    }
-}
+            let shutdown = Controller::shutdown_handle();
 
-#[derive(Debug)]
-struct SystemInfo {
-    system: System,
-    last_refresh: Instant,
-    refresh_interval: Duration,
-}
+            while shutdown.get().is_none() {
+                let _ = update_tx.send(StatusUpdate::new(relay_statsd::metric!(
+                    timer(RelayTimers::HealthCheckDuration),
+                    type = "readiness",
+                    { self.check_readiness().await }
+                )));
 
-impl SystemInfo {
-    /// Creates a new [`SystemInfo`] to query system information.
-    ///
-    /// System information updates are debounced with the passed `refresh_interval`.
-    pub fn new(refresh_interval: Duration) -> Self {
-        let system = System::new_with_specifics(
-            RefreshKind::new().with_memory(MemoryRefreshKind::everything()),
-        );
-
-        Self {
-            system,
-            last_refresh: Instant::now(),
-            refresh_interval,
-        }
-    }
-
-    /// Current snapshot of system memory.
-    ///
-    /// On Linux systems it uses the cgroup limits to determine used and total memory,
-    /// if available.
-    pub fn memory(&mut self) -> Memory {
-        self.refresh();
-
-        // Use the cgroup if available in case Relay is running in a container.
-        if let Some(cgroup) = self.system.cgroup_limits() {
-            Memory {
-                used: cgroup.total_memory.saturating_sub(cgroup.free_memory),
-                total: cgroup.total_memory,
-            }
-        } else {
-            Memory {
-                used: self.system.used_memory(),
-                total: self.system.total_memory(),
+                tokio::time::sleep(check_interval).await;
             }
-        }
-    }
 
-    fn refresh(&mut self) {
-        if self.last_refresh.elapsed() >= self.refresh_interval {
-            self.system
-                .refresh_memory_specifics(MemoryRefreshKind::new().with_ram());
+            // Shutdown marks readiness health check as unhealthy.
+            update_tx.send(StatusUpdate::new(Status::Unhealthy)).ok();
```

Comment:
```suggestion
            let _ = update_tx.send(StatusUpdate::new(Status::Unhealthy));
```

Converting to an option just to get rid of the `#[must_use]` is a smell imo, just ignore the error explicitly.

Reply:
We use both extensively in the code base, `.ok()` is definitely a widely used pattern